src/tools/rpmemd/rpmemd_log.c

```diff
@@ -195,8 +195,11 @@ rpmemd_log(enum rpmemd_log_level level, const char *fname, int lineno,
 	size_t cnt = 0;
 	int ret;
 	if (fname) {
+		char *basename_fname = strdup(fname);
+		basename_fname = basename(basename_fname);
 		ret = snprintf(&buff[cnt], RPMEMD_MAX_MSG - cnt,
-				"[%s:%d] ", basename(fname), lineno);
+				"[%s:%d] ", basename_fname, lineno);
+		free(basename_fname);
```

Comment:
1) On glibc basename doesn't modify its argument when _GNU_SOURCE macro is defined (and this is what we do at the beginning of this file).
This is a logging function, which may be called quite frequently, so mallocing every time may be quite costly. If you don't want to simulate what _GNU_SOURCE version does, I'd prefer to have this part abstracted.
For example by introducing libc-dependent version of os_basename/os_basename_free that either allocates/frees a buffer or is empty.

2) Even without the above this change doesn't seem correct - you are freeing value returned by basename, instead of the allocated buffer. This could fail if basename would return pointer to the middle of the path.